### PR TITLE
feat: 新增华为 Welink TXT 格式解析器

### PR DESCRIPTION
**Pull request by l17728**
## Summary

新增自定义 TXT 格式解析器，适配 **华为 Welink** 导出的聊天记录。

支持格式：`昵称(ID)\tYYYY-MM-DD HH:mm:ss\n消息内容`

### 新功能
- **custom-txt parser**: 灵活的 TXT 格式解析器，支持自定义格式导入
- **e2e 测试框架**: Playwright 端到端测试框架搭建

### 测试覆盖
- **Unit Tests (UT)**: `custom-txt.test.ts` - 正常路径、边界条件、异常处理
- **Integration Tests (IT)**: `custom-txt.integration.test.ts` - 解析器与导入流程集成测试
- **System Tests (ST)**: `tests/system/custom-txt-import.test.ts` - 完整导入流程测试
- **e2e Tests**: `tests/e2e/custom-txt-import.test.ts` - 用户界面导入功能测试

### 文档更新
- `AGENTS.md`: 测试覆盖率要求和开发指南

### 测试样本
- `qq_chat.txt`: QQ 聊天样本
- `test-chat-sample.txt`: 通用测试样本

## Review 问题已解决

根据 Codex 自动 review 的建议，已修复以下问题：

### ✅ CDP 端口占用问题 (P2)
- **原问题**: 端口 9222 被占用时只警告并继续，可能连接到错误进程
- **解决方案**: 
  - 默认动态分配可用端口（避免冲突）
  - 指定端口被占用时快速失败（抛出 Error）
  - 文件: `tests/e2e/helpers/app-launcher.js`

### ✅ 消息头尾部空格问题 (P2)
- **原问题**: 消息头末尾有空格时无法匹配，导致解析失败
- **解决方案**: 
  - 使用 `rawLine.trimEnd()` 去除行尾空白后再匹配正则
  - 文件: `electron/main/parser/formats/custom-txt.ts`

## Test Coverage

所有新功能包含四层测试：
- ✅ Normal path (Happy Path)
- ✅ Boundary conditions
- ✅ Error handling
- ✅ Null/empty data handling

## Files Changed

- 23 files changed
- 6717+ lines added

### CODE REVIEW

**Comment by hellodigua** [comment]
我想询问，华为Welink官方是支持导出的吗，是任何用户都可以导出，还是仅管理员可导出，导出方式的步骤？这对更新用户导出文档比较重要

另外，custom-txt 这个命名令人疑惑，它从语义上太过通用了，建议改名为 welink-txt，后续如果有json等格式，即命名为 welink-json等。

同时新增的 Playwright 端到端测试框架搭建，建议拆分为一个独立的commit

**Comment by l17728** [comment]
支持导出，任何用户都可以导出。导出的步骤是点击发送消息框右上角的**聊天记录**按钮，在弹出的面板右上方点击**高级搜索**，在弹出的聊天记录对话框右侧有导出按钮，选择txt格式。
好的，我修改一些名字，分成独立的commit

File: qq_chat.txt
Comment (hellodigua) [inline qq_chat.txt]: 这个文件可以移除

Comment (hellodigua) [inline qq_chat.txt]: 这个文件也可以移除

File: electron/main/parser/formats/index.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ import whatsappNativeTxt from './whatsapp-native-txt'
 import qqNativeTxt from './qq-native-txt'
 import instagramNative from './instagram-native'
 import lineNativeTxt from './line-native-txt'
+import customTxt from './custom-txt'
```
[the remark this answers is not in the document]
Comment (l17728) [reply electron/main/parser/formats/index.ts]: 👍

File: tests/e2e/specs/app-launcher-port.spec.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  test('连续启动多个实例应该分配不同端口', async () => {
+    console.log('[E2E] 测试多实例端口分配')
+
+    const apps = []
+    const ports = []
+
+    try {
+      // 连续启动 3 个实例
+      for (let i = 0; i < 3; i++) {
+        const app = await launchApp()
```
[the remark this answers is not in the document]
Comment (l17728) [reply tests/e2e/specs/app-launcher-port.spec.js]: 已修改

File: electron/main/parser/formats/welink-txt.ts [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+  fileStream.on('data', (chunk: string | Buffer) => {
+    bytesRead += typeof chunk === 'string' ? Buffer.byteLength(chunk) : chunk.length
+  })
+
+  let lineNum = 0
+  for await (const rawLine of rl) {
+    lineNum++
+
+    // 去除行尾空白字符（处理尾随空格/制表符）
+    const line = rawLine.trimEnd()
```
[the remark this answers is not in the document]
Comment (l17728) [reply electron/main/parser/formats/welink-txt.ts]: 已修改

File: electron/main/parser/formats/index.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (l17728) [reply electron/main/parser/formats/index.ts]: 已修改

**Comment by hellodigua** [comment]
请看上方review的3个comment，另外可以squash到两个commit吗，直接reset commit后然后重新提交也不是不行

**Comment by l17728** [comment]
好，我明天处理一下

File: tests/e2e/specs/welink-txt-import.spec.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    expect(result.success).toBe(true)
+    expect(result.result.success).toBe(true)
+    expect(result.result.sessionId).toBeTruthy()
+
+    // 验证消息数量
+    expect(result.result.diagnostics.messagesWritten).toBe(54)
+    expect(result.result.diagnostics.messagesSkipped).toBe(0)
+
+    // 验证格式识别
+    expect(result.result.diagnostics.detectedFormat).toContain('自定义 TXT')
```
[the remark this answers is not in the document]
Comment (l17728) [reply tests/e2e/specs/welink-txt-import.spec.js:109]: 已修改

File: electron/main/parser/formats/welink-txt.integration.test.ts [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+    const filePath = createTempFile(content)
+    console.log('[Test] 测试非 welink-txt 格式')
+
+    try {
+      const feature = detectFormat(filePath)
+      console.log('[Test] 识别结果:', feature?.id, feature?.name)
+
+      // 应该被识别为 QQ 格式而不是 custom-txt
+      if (feature) {
+        assert.notEqual(feature.id, 'custom-txt', 'QQ 格式不应该被识别为 custom-txt')
```
[the remark this answers is not in the document]
Comment (l17728) [reply electron/main/parser/formats/welink-txt.integration.test.ts:168]: 已修改
